### PR TITLE
Add support for Fortanix DSM signer

### PR DESCRIPTION
**Pull request by mzohreva**
I'll need to update documentation to reflect the new signing provider, but wanted to get some early feedback first.

### CODE REVIEW

**Comment by tony-iqlusion** [comment]
Mind rebasing? We just cut a new release and can start looking at this

**Comment by mzohreva** [comment]
@tony-iqlusion any progress on reviewing this? Let me know if you need more information about Fortanix DSM.

**Comment by tony-iqlusion** [comment]
@mzohreva it looks like the checks didn't run. Can you try doing a quick `git commit --amend` and `push --force-with-lease` and see if that kicks them off?

**Comment by mzohreva** [comment]
@tony-iqlusion just did what you asked, I see no change. In the **Checks** tab I see this: `This workflow is awaiting approval from a maintainer in #469`

**Comment by tony-iqlusion** [comment]
@mzohreva I have a button on my side now, which I just hit, and the workflow is running

**Comment by mzohreva** [comment]
Seems like `ed25519 1.4.0` is yanked, I'll update

**Comment by tony-iqlusion** [comment]
Looks like a clippy failure

**Comment by mzohreva** [comment]
@tony-iqlusion just addressed the clippy issues.

**Comment by tony-iqlusion** [comment]
Thanks! The build is finally green. Will take a more in-depth look tomorrow.

File: src/keyring/providers/fortanixdsm.rs [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+        if spki.algorithm.parameters != None {
+            // TODO: once/if https://github.com/RustCrypto/formats/issues/354 is addressed we should use that error variant.
+            return Err(SpkiError::KeyMalformed);
+        }
+
+        Ed25519::from_bytes(spki.subject_public_key)
+            .map_err(|_| SpkiError::KeyMalformed)
+            .map(Ed25519PublicKey)
+    }
+}
```
Comment (tony-iqlusion) [inline src/keyring/providers/fortanixdsm.rs:233]: Likewise most of this is available as [`ed25519::pkcs8::PublicKeyBytes`](https://docs.rs/ed25519/latest/ed25519/pkcs8/struct.PublicKeyBytes.html)

**Pull request merged**